src/Controls/src/Core/BindingFactory.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using Microsoft.Maui.Controls.Internals;
+
+namespace Microsoft.Maui.Controls
+{
+    public static class BindingFactory
+    {
+        /// <summary>
+        /// This factory method was added to simplify creating TypedBindingBase instances from lambda getters.
+        /// </summary>
+        /// <typeparam name="TSource">The source type.</typeparam>
+        /// <typeparam name="TProperty">The property type.</typeparam>
+        /// <param name="getter">An getter method used to retrieve the source property.</param>
+        /// <param name="mode">The binding mode. This property is optional. Default is <see cref="F:Microsoft.Maui.Controls.BindingMode.Default" />.</param>
+        /// <param name="converter">The converter. This parameter is optional. Default is <see langword="null" />.</param>
+        /// <param name="converterParameter">An user-defined parameter to pass to the converter. This parameter is optional. Default is <see langword="null" />.</param>
+        /// <param name="stringFormat">A String format. This parameter is optional. Default is <see langword="null" />.</param>
+        /// <param name="source">An object used as the source for this binding. This parameter is optional. Default is <see langword="null" />.</param>
+        /// <param name="fallbackValue">The value to use instead of the default value for the property, if no specified value exists.</param>
+        /// <param name="targetNullValue">The value to supply for a bound property when the target of the binding is <see langword="null" />.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static TypedBinding<TSource, TProperty> Create<TSource, TProperty>(
```

Comment:
Adding this method to `Binding` is problematic because we need to keep the class annotated with `[RequiresUnreferencedCode]`. One more option would be to rename `TypedBinding<TSource, TProperty>` to `Binding<TSource, TProperty>` and have `Binding<TSource, TProperty>.Create(Func<TSource, TProperty> getter, ...)`

Reply:
I changed the return type to `BindingBase` and I moved this under the `Binding` class with an `[UnconditionalSuppressMessage]`. I think this is a valid case for the suppression.